src/checksum.rs

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::error;
+use std::fmt;
+
+pub(crate) const CHECKSUM_LEN: usize = 8;
+
+pub(crate) type Checksum = u64;
+
+#[derive(Clone, Debug)]
+pub struct ChecksumError;
```

Comment:
Note: this `pub` is unreachable because `ironfish_frost::checksum` is private. Either re-export this error at the root of the crate, or make the module `pub` too.